services/web/server/src/simcore_service_webserver/projects/projects_nodes_utils.py

```diff
@@ -0,0 +1,130 @@
+import logging
+from typing import Any, Coroutine, Optional
+
+from aiohttp import web
+from models_library.errors import ErrorDict
+from pydantic.types import PositiveInt
+from servicelib.logging_utils import log_decorator
+from servicelib.utils import logged_gather
+from collections import deque
+from models_library.users import UserID
+from models_library.projects import ProjectID
+from .projects_utils import get_frontend_node_outputs_changes
+from servicelib.utils import fire_and_forget_task
+from . import projects_api
```

Comment:
actually a little bit weird that a utils calls the project_api... maybe that is one part of your issue with circular dependencies

Reply:
No, I've removed the circular dependency, this ok now. This is no longer called from the `projects_db` but from the `projects_handlers_crud.py`. PC actually pointed out the issue.